react/leanShipping.js

```diff
@@ -294,35 +295,52 @@ export function getCombinedOption(
   )
 }
 
+// Get SLA's able to deliver all cart
+function getConsistentSlas(items) {
+  let consistentSlas = []
+
+  for (let slas of items) {
+    if (consistentSlas.length === 0) consistentSlas = slas.map(sla => sla.id)
+    consistentSlas = consistentSlas.filter(slaId => slas.some(sla => sla.id === slaId))
+  }
+
+  return consistentSlas
+}
+
 function getMinSlaBy(items, property) {
+  const consistentSlas = getConsistentSlas(items)
+
+  for (let i = 0; i < items.length; i += 1) {
+    // Ordering by consistent SLA's makes it our tiebreaker
+    items[i] = orderBy(items[i], sla => consistentSlas.some(slaId => sla.id !== slaId))
```

Comment:
in order to avoid mutating the function argument, can you rewrite this loop to a `map()` instead?

Reply:
Sure. I did it with a map first, but I was struggling to debug its iterations (in the end, it had nothing to do with my difficulties... I was just blind to some typos :harold-pain:).